subsys/bluetooth/mesh/shell.c

```diff
@@ -1721,15 +2018,94 @@ static int cmd_mod_pub(const struct shell *shell, size_t argc, char *argv[])
 	}
 }
 
+static int mod_pub_va_set(const struct shell *shell, uint16_t addr, uint16_t mod_id,
+			  uint16_t cid, char *argv[])
+{
+	struct bt_mesh_cfg_mod_pub_va pub_va;
+	uint8_t status, count;
+	uint8_t label[16];
+	uint16_t interval;
+	size_t len;
+	int err;
+
+	len = hex2bin(argv[0], strlen(argv[0]), label, sizeof(label));
+	(void)memset(label + len, 0, sizeof(label) - len);
+	pub_va.label = (const uint8_t *)&label;
+	pub_va.pub.app_idx = strtoul(argv[1], NULL, 0);
+	pub_va.pub.cred_flag = str2bool(argv[2]);
+	pub_va.pub.ttl = strtoul(argv[3], NULL, 0);
+	pub_va.pub.period = strtoul(argv[4], NULL, 0);
+
+	count = strtoul(argv[5], NULL, 0);
+	if (count > 7) {
+		shell_print(shell, "Invalid retransmit count");
+		return -EINVAL;
+	}
+	interval = strtoul(argv[6], NULL, 0);
+	if (interval > (31 * 50) || (interval % 50)) {
+		shell_print(shell, "Invalid retransmit interval %u", interval);
+		return -EINVAL;
+	}
+
+	pub_va.pub.transmit = BT_MESH_PUB_TRANSMIT(count, interval);
+
+	if (cid == CID_NVAL) {
+		err = bt_mesh_cfg_mod_pub_va_set(net.net_idx, net.dst, addr,
+						 mod_id, &pub_va, &status);
+	} else {
+		err = bt_mesh_cfg_mod_pub_va_set_vnd(net.net_idx, net.dst, addr,
+						     mod_id, cid, &pub_va, &status);
+	}
+
+	if (err) {
+		shell_error(shell, "Model Publication Set failed (err %d)",
+			    err);
+		return 0;
+	}
+
+	if (status) {
+		shell_print(shell, "Model Publication Set failed "
+			    "(status 0x%02x)", status);
+	} else {
+		shell_print(shell, "Model Publication successfully set");
+	}
+
+	return 0;
+}
+
+static int cmd_mod_pub_va(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint16_t addr, mod_id, cid;
+
+	if (argc < 10) {
+		return -EINVAL;
+	}
+
+	addr = strtoul(argv[1], NULL, 0);
+	mod_id = strtoul(argv[2], NULL, 0);
+
+	argc -= 3;
+	argv += 3;
```

Comment:
This seems like a potential future bug, why not keep do `argv[3]` below?

Reply:
?